### ISSUE TITLE
Bug: Invalid JSON when no web server is found

### ISSUE THREAD

**Issue by JohannesZahn** [sullo/nikto#740]
### Expected behavior
When I run
`./nikto.pl -h www.example.com -p 8080 -Tuning 1,2,3,5,7,b -o out.json `

and no web server is found, the output is not valid json. Overall I would expect to get something like this

```json
{
  "host": "example.com",
  "ip": "...",
  "port": "8080",
  "banner": "",
  "vulnerabilities": [
    {
      "id": "000029",
      "OSVDB": "0",
      "method": "GET",
      "url": "/",
      "msg": "No web server found on www.example.com:8080"
    }]
}
```
### Actual behavior
The following output is produced 
```json
{
  "id": "000029",
  "OSVDB": "0",
  "url":"/",
  "msg":"No web server found on www.example.com:8080"
}
}
```
### Steps to reproduce

1. run ./nikto.pl -h www.example.com -p 8080 -Tuning 1,2,3,5,7,b -o out.json
2. Inspect out.json file

### Nikto version
v2.1.6 (master)

Run:

```
./nikto.pl -Version
```

and paste the output here.

---------------------------------------------------------------------------
Nikto Versions
---------------------------------------------------------------------------
File                               Version      Last Mod
-----------------------------      --------     ----------
Nikto main                         2.1.6
LibWhisker                         2.5
db_404_strings                     2.003        
db_content_search                  2.000        
db_dictionary                      1.0          
db_dir_traversal                   2.1.6        
db_domino                          2.1.6        
db_drupal                          1.00         
db_embedded                        2.004        
db_favicon                         2.010        
db_headers                         2.008        
db_httpoptions                     2.002        
db_multiple_index                  2.005        
db_outdated                        2.017        
db_parked_strings                  2.001        
db_realms                          2.002        
db_server_msgs                     2.006        
db_tests                           2.021        
db_variables                       2.004        
nikto_apache_expect_xss.plugin     2.04         
nikto_apacheusers.plugin           2.06         
nikto_auth.plugin                  2.04         
nikto_cgi.plugin                   2.06         
nikto_clientaccesspolicy.plugin    1.00         
nikto_content_search.plugin        2.05         
nikto_cookies.plugin               2.05         
nikto_core.plugin                  2.1.5        
nikto_dictionary_attack.plugin     2.04         
nikto_dir_traversal.plugin         2.1.6        
nikto_dishwasher.plugin            2.20         
nikto_docker_registry.plugin       2.20         
nikto_domino.plugin                2.1.6        
nikto_drupal.plugin                1.00         
nikto_embedded.plugin              2.07         
nikto_favicon.plugin               2.09         
nikto_fileops.plugin               1.00         
nikto_headers.plugin               2.11         
nikto_httpoptions.plugin           2.10         
nikto_ms10_070.plugin              1.00         
nikto_msgs.plugin                  2.07         
nikto_multiple_index.plugin        2.03         
nikto_negotiate.plugin             2.00         
nikto_origin_reflection.plugin     2.01         
nikto_outdated.plugin              2.09         
nikto_parked.plugin                2.00         
nikto_paths.plugin                 2.00         
nikto_put_del_test.plugin          2.04         
nikto_report_csv.plugin            2.07         
nikto_report_html.plugin           2.06         
nikto_report_json.plugin           2.00         
nikto_report_nbe.plugin            2.02         
nikto_report_sqlg.plugin           2.00         
nikto_report_text.plugin           2.05         
nikto_report_xml.plugin            2.06         
nikto_robots.plugin                2.06         
nikto_shellshock.plugin            2.01         
nikto_siebel.plugin                1.00         
nikto_sitefiles.plugin             2.00         
nikto_ssl.plugin                   2.01         
nikto_strutshock.plugin            2.01         
nikto_tests.plugin                 2.04         
---------------------------------------------------------------------------
### Further technical info

E.g. you can obtain Nikto debug output by running `-D D` and redirecting to a file.
You may also scrub the output of hostnames and IPs by specifying `-D DS`.

**Comment by sullo** [comment]
This should be fixed in  the 2.5.0 branch. If you can validate taut would he helpful. 

Thanks
Sullo 

>

**Comment by JohannesZahn** [comment]
I already tried that out and I can confirm it is fixed in 2.5.0 :).

If I can find a simple solution for this I will create a PR for v2.1.6 also.

**Comment by JohannesZahn** [comment]
You are right, this does not work for multiple hosts, we are currently trying out v2.5 and I will reopen if it does not work for us.
